<a href="https://colab.research.google.com/github/dvisionst/Project_Two/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [4]:
df = pd.read_csv("/content/neo.csv")
df.head()


id                 name  est_diameter_min  est_diameter_max  \
0  2162635  162635 (2000 SS164)          1.198271          2.679415   
1  2277475    277475 (2005 WK4)          0.265800          0.594347   
2  2512244   512244 (2015 YE18)          0.722030          1.614507   
3  3596030          (2012 BV13)          0.096506          0.215794   
4  3667127          (2014 GE35)          0.255009          0.570217   

   relative_velocity  miss_distance orbiting_body  sentry_object  \
0       13569.249224   5.483974e+07         Earth          False   
1       73588.726663   6.143813e+07         Earth          False   
2      114258.692129   4.979872e+07         Earth          False   
3       24764.303138   2.543497e+07         Earth          False   
4       42737.733765   4.627557e+07         Earth          False   

   absolute_magnitude  hazardous  
0               16.73      False  
1               20.00       True  
2               17.83      False  
3               22.20      False  
4               20.09       True

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90836 entries, 0 to 90835
Data columns (total 10 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   id                  90836 non-null  int64  
 1   name                90836 non-null  object 
 2   est_diameter_min    90836 non-null  float64
 3   est_diameter_max    90836 non-null  float64
 4   relative_velocity   90836 non-null  float64
 5   miss_distance       90836 non-null  float64
 6   orbiting_body       90836 non-null  object 
 7   sentry_object       90836 non-null  bool   
 8   absolute_magnitude  90836 non-null  float64
 9   hazardous           90836 non-null  bool   
dtypes: bool(2), float64(5), int64(1), object(2)
memory usage: 5.7+ MB


1) Source of data
*NASA via Kaggle*

2) Brief description of data
*This data set shows a large number of near earth asteroids, and their attributes. It displays size, velocity, disstance and if it's hazardous*

3) What is the target?
*The 'hazardous' column is the target*

4) Is this a classification or regression problem?
*This is a classification problem*

5) How many features?
*There are 10 columns with one being an ID column.*

6) How many rows of data.
*90,835 rows*

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

*This is a very large data set, hypertuning parameters and going through differnet models will be a more time consuming process. *

I chose this data set because I have always been interested in space.I'm extremely curious to know what data we have collected in terms of near earth objects. I think that as a society any sizable asteriod impact would be apocalyptic. This makes identifying as many hazardous objects near Earth a very important task. The next would be how to prevent the collision. For now i want to explore this data set for myself. 

In [10]:
df = pd.read_csv("/content/Environment_Temperature_change_E_All_Data_NOFLAG.csv", encoding='cp1252')
df.head()

Area Code         Area  Months Code    Months  Element Code  \
0          2  Afghanistan         7001   January          7271   
1          2  Afghanistan         7001   January          6078   
2          2  Afghanistan         7002  February          7271   
3          2  Afghanistan         7002  February          6078   
4          2  Afghanistan         7003     March          7271   

              Element Unit  Y1961  Y1962  Y1963  ...  Y2010  Y2011  Y2012  \
0  Temperature change   °C  0.777  0.062  2.744  ...  3.601  1.179 -0.583   
1  Standard Deviation   °C  1.950  1.950  1.950  ...  1.950  1.950  1.950   
2  Temperature change   °C -1.743  2.465  3.919  ...  1.212  0.321 -3.201   
3  Standard Deviation   °C  2.597  2.597  2.597  ...  2.597  2.597  2.597   
4  Temperature change   °C  0.516  1.336  0.403  ...  3.390  0.748 -0.527   

   Y2013  Y2014  Y2015  Y2016  Y2017  Y2018  Y2019  
0  1.233  1.755  1.943  3.416  1.201  1.996  2.951  
1  1.950  1.950  1.950  1.950  1.950  1.950  1.950  
2  1.494 -3.187  2.699  2.251 -0.323  2.705  0.086  
3  2.597  2.597  2.597  2.597  2.597  2.597  2.597  
4  2.246 -0.076 -0.497  2.296  0.834  4.418  0.234  

[5 rows x 66 columns]

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9656 entries, 0 to 9655
Data columns (total 66 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     9656 non-null   int64  
 1   Area          9656 non-null   object 
 2   Months Code   9656 non-null   int64  
 3   Months        9656 non-null   object 
 4   Element Code  9656 non-null   int64  
 5   Element       9656 non-null   object 
 6   Unit          9656 non-null   object 
 7   Y1961         8287 non-null   float64
 8   Y1962         8322 non-null   float64
 9   Y1963         8294 non-null   float64
 10  Y1964         8252 non-null   float64
 11  Y1965         8281 non-null   float64
 12  Y1966         8364 non-null   float64
 13  Y1967         8347 non-null   float64
 14  Y1968         8345 non-null   float64
 15  Y1969         8326 non-null   float64
 16  Y1970         8308 non-null   float64
 17  Y1971         8303 non-null   float64
 18  Y1972         8323 non-null 

Second choice: dataset 2

1) Source of data
*NASA via Kaggle*

2) Brief description of data
*The monthly, seasonal and annual mean temperature for every country from 1961-2019*

3) What is the target?
*The target is the 2019 year column in order to predict the mean temperature from all previous years*

4) Is this a classification or regression problem?
*This is a regression problem, using data for every country going back to 1961, in order to predict the mean temperature*

5) How many features?
*It has 66 columns but the majority of these are dedicated to different years*

6) How many rows of data.
*There are 9656 rows of data*

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

*This will be challenging because of how the data is formatted. Will really have to organize it the way I need it using filters. Substantial data prepping. *